### PR TITLE
To Do list Interactive List

### PR DESCRIPTION
**Pull request by Jord4N-AC**
In this project, I added some functionality to my application to make it interactive. The user will also be able to mark task completion by selecting the corresponding checkbox (or undo it by unchecking the checkbox). The updated tasks list gets stored in local storage.

### Project requirements & Feautures

- Add a new JavaScript file and import it as a module, it contains methods related to the status updates (completed: true/false)
- Add event listener to the checkbox
- Update items object's value for completed key upon user actions
- Implement a function for the "Clear all completed" button
- Store the updated array of items in local storage, so the user gets the correct list values when the page reloads.

#### Additional

- When the user adds a task a message saying 'Added successfully' appears, the page scrolls to show the new task and it gest highlight
- When the user adds a task that already exists a message saying 'Task repeated' appears, the page scrolls to show the repeated task and it gest highlight
- All text that the user inputs is trimmed (remove extra spaces from sides and middle), and empty text is not allowed
- When the user edits a task and adds empty text it returns the latest task
- You can add a new task, by pressing enter on input, pressing <kbd>↲</kbd> enter on Add button, or click on Add <kbd>↲</kbd> button
- You can mark a task as completed, clicking on the checkbox or pressing enter on the checkbox (This is when you want to navigate using Tab <kbd>↹</kbd> bey)

### General Requirements

- Check linter errors
- Use the correct GitHub flow
- Document my work in a professional way
- Follow best practices for HTML, CSS & JavaScript
- Open a pull request and ask for a review

### CODE REVIEW

File: src/modules/remove_completed.js [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+export default function removeAllCompleted() {
+  while (completedElements.length !== 0) {
+    taskArr.splice(completedElements[0].id, 1);
+    completedElements[0].remove();
+
+    taskArr.forEach((task, i) => {
+      task.index = i;
+      taskList.children[i].id = i;
+    });
+  }
```
Comment (elmar8287) [inline src/modules/remove_completed.js]: - [x] Please, consider using the `filter()` method implementation of a function for the "Clear all completed" button, as it is required in the project. With the `forEach` version, need to analyze that there is an empty array defined first and then elements are added to it based on the condition. With the `filter()` function, we only need to concentrate on the test function logic. You can read more [here](https://medium.com/front-end-weekly/stop-array-foreach-and-start-using-filter-map-some-reduce-functions-298b4dabfa09) about `filter` and `map` methods

**Review by elmar8287 — CHANGES_REQUESTED** [review]
Hi, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

- Perfect design and functionality 👍 
- Informative PR and Readme file
- Good commits history 👍 

## Required Changes ♻️

- [x] Check the comments under the review.

## Optional suggestions

- N/A

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/modules/remove_completed.js (hunk 1 above)
Comment (Jord4N-AC) [reply src/modules/remove_completed.js]: @elmar8287
the function forEach() method is to update the ids when removing the completed elements, I don't  know if it's really necessary that I used forEach() method

Comment (elmar8287) [reply src/modules/remove_completed.js]: > 

Dear @Jord4N-AC could you implement the filter() method as required in the project? I think you can do it easily 👍

**Review by zieeco — APPROVED** [review]
# status: APPROVED :green_circle: :100: 

Hi, @Jord4N-AC 


68747470733a2f2f6d656469612e67697068792e636f6d2f6d656469612f78543078657a514755357843444a754350652f67697068792e676966


Good job so far 👍 Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 


## Highlights ⚡ 
✅ Nice design 🔥 
✅ Linters are passing 👏 


## Optional suggestions
- N/a


_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**